lib/spack/docs/packaging_guide.rst

```diff
@@ -2343,21 +2348,24 @@ requires Python 2, you can similarly leave out the lower bound:
 Notice that we didn't use ``@:3``. Version ranges are *inclusive*, so
 ``@:3`` means "up to and including any 3.x version".
 
-What if a package can only be built with Python 2.7? You might be
-inclined to use:
+If we require Python 3 exactly, we can simply write:
 
 .. code-block:: python
 
-   depends_on('python@2.7')
+   depends_on('python@3')
 
-However, this would be wrong. Spack assumes that all version constraints
-are exact, so it would try to install Python not at ``2.7.18``, but
-exactly at ``2.7``, which is a non-existent version. The correct way to
-specify this would be:
+since we don't care about the latter digits. You can think of this as a
+short-hand for ``@3:3``. Among equivalent specs, usually the most succinct is
+preferred.
 
-.. code-block:: python
+Note that if a version part is alphabetic, it is considered to precede
```

Comment:
This is already explained in https://spack.readthedocs.io/en/latest/packaging_guide.html#version-comparison, I don't think it should be part of ranges